### PR TITLE
Forward notifications from the UI process to the WebContent process

### PR DESCRIPTION
**Pull request by pvollan**
#### b35a3a5d21193edf78d8c645f02045f8d3af039a
<pre>
Forward notifications from the UI process to the WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=275874">https://bugs.webkit.org/show_bug.cgi?id=275874</a>
<a href="https://rdar.apple.com/129602281">rdar://129602281</a>

Reviewed by Sihui Liu and Ben Nham.

When a notification is being observed in the WebContent process, an OS fault is generated, unless this
notification has been added to an allow list in the entitlements. The purpose of this telemetry is to
detect which notifications we need to forward from the UI process. This patch adds a set of notifications
that has generated faults to the allow list. The notifications are also added to the list of forwarded
notifications, and to the sandbox so that they can be reposted in the WebContent process.

It is important to remove these OS faults by adding the notifications to these allow lists, since they
can be costly in terms of CPU usage.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::registerNotificationObservers):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/280421@main">https://commits.webkit.org/280421@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/8c03c887ecaf5b4b1e3b91c332527519fe11a4ba

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56595 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35922 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9068 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60209 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7036 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43545 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7227 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45845 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4923 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58625 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33771 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48843 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26706 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/56121 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30552 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6039 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/52509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6446 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61892 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/507 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6551 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53102 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/508 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48902 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53099 "Found 3 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed, /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure, /WebKitGTK/TestInspectorServer:/webkit/WebKitWebInspectorServer/http-test-page-list (failure)") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/12522 "The change is no longer eligible for processing.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/435 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8407 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31750 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32836 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33920 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32582 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm [hunk 1]
```diff
@@ -766,6 +775,10 @@ static void logProcessPoolState(const WebProcessPool& pool)
         "com.apple.system.DirectoryService.InvalidateCache.user"_s,
 #else
         // EMBEDDED_FORWARDED_NOTIFICATIONS
+        "com.apple.ManagedConfiguration.profileListChanged"_s,
+        "com.apple.managedconfiguration.passcodechanged"_s,
+        "com.apple.managedconfiguration.restrictionchanged"_s,
+        "com.apple.managedconfiguration.settingschanged"_s,
```
Comment (szewai) [inline Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:781]: My understanding is these should be synced with EMBEDDED_FORWARDED_NOTIFICATIONS in process-entitlements.sh too, but this patch does not add them there, so it seems wrong.

Also, can we keep this patch to only adding new notifications? It's a bit unclear what notifications are new in this patch...

Comment (pvollan) [reply Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:781]: Ah, that's a good catch, will fix!

I will also move the notifications to common.sb in a separate patch, in order to make it more clear what the new notifications are.

Thanks for reviewing!

**Review by bnham — APPROVED** [review]

**Comment by pvollan** [comment]
Thank you for reviewing!

File: Source/WebKit/Scripts/process-entitlements.sh [hunk 2]
```diff
@@ -281,21 +294,29 @@ function notify_entitlements()
...
+            "com.apple.accessibility.cache.quick.speak"
             "com.apple.accessibility.cache.reduce.motion"
+            "com.apple.accessibility.cache.reduce.motion.reduce.slide.transitions"
             "com.apple.accessibility.cache.reduce.motioncom.apple.WebKit.WebContent"
+            "com.apple.accessibility.cache.speak.this"
             "com.apple.accessibility.cache.speech.settings.disabled.by.mc"
             "com.apple.accessibility.cache.switch.control"
             "com.apple.accessibility.cache.vot"
             "com.apple.accessibility.cache.zoom"
+            "com.apple.accessibility.cache.*"
```
Comment (szewai) [inline Source/WebKit/Scripts/process-entitlements.sh:320]: If wildcard match is supported, why do we need to specify each notification above?

Comment (pvollan) [reply Source/WebKit/Scripts/process-entitlements.sh:320]: That's a good point! I'll add a comment that we can remove the other notification names once wildcard is supported everywhere.

Thanks for reviewing!

**Review by szewai — APPROVED** [review]

**Pull request merged**